### PR TITLE
Refactored sti-criteria-2

### PR DESCRIPTION
**Pull request by ashih2018**
Fixed up comments made by @david-yz-liu in #4749 

## Motivation and Context
Refactor code based on comments made so #4749 can be merged into master, introducing the STI criteria database structure into master.

## Your Changes
Edited various code in the codebase for models and tests.

**Type of change** (select all that apply):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring (internal change to codebase, without changing functionality)
- [ ] Test update (change that modifies or updates tests only)
- [ ] Other (please specify): 

## Testing
- Making sure Travis doesn't raise any errors should be sufficient.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments. 
- [x] I have verified that the TravisCI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [ ] I have added tests for my changes. 
- [ ] I have updated the Changelog.md file. 
- [ ] I have described any required documentation changes below.

### CODE REVIEW

**Comment by ashih2018** [comment]
Can these usages of `criterion_type` be deleted as well?
https://github.com/MarkUsProject/Markus/blob/3476f099caba2a7494582df4bd672fb235242795/app/assets/javascripts/Components/Result/result.jsx#L106
https://github.com/MarkUsProject/Markus/blob/3476f099caba2a7494582df4bd672fb235242795/app/assets/javascripts/Components/Result/result.jsx#L220
The second one in particular seems important and probably shouldn't be deleted.

File: app/assets/javascripts/Components/Result/result.jsx [hunk 1]
```diff
@@ -364,7 +364,7 @@ class Result extends React.Component {
     });
   };
 
-  destroyMark = (criterion_type, criterion_id) => {
+  destroyMark = (criterion_id) => {
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/Result/result.jsx:367]: Did you look for where this method is called? It's very suspicious to change the signature of a function but not change anywhere else in the code.

**Comment by david-yz-liu** [comment]
You can remove both of those instances of `criterion_type`. For the first one, look for where the data is being passed in the controller method.

**Comment by ashih2018** [comment]
@david-yz-liu When I checked out the first instance of `criterion_type`, I realized that was likely a typo as we save `old_mark` into the state but the only time we reference it is passing it down to our components as `old_marks`.
I looked at `results_controller.rb` at where we are updating `old_marks` and it turns out that I previously removed all usages of the `criterion_type` in earlier refactoring.

**Pull request merged**